docs/explanation/e-hardening.md

```diff
@@ -0,0 +1,119 @@
+# Security Hardening Guide
+
+This document provides all the guidance and instructions needed to achieve 
+a secure deployment of the Charmed Kafka product, including setting up and managing a secure environment.
+The document is divided into the following sections:
+
+1. Environment, outlining the recommendation for deploying a secure environment
+2. Applications, outlining the product features that enable a secure deployment of a Kafka cluster
+3. Additional resources, providing any further information about security and compliance
+
+## Environment
+
+The environment where applications operate can be divided in three components:
```

Comment:
We need to either add OS back in the list below, or replace `three` with `two`.
I'd prefer to add OS to the Environment, but see my comment below about talking specifically about the Host machine's OS and VM's OS.